misc/generate-routes-file-for-dynamic-routes.js

```diff
@@ -0,0 +1,120 @@
+/* The purpose of this script is to generate the `routes.txt` file that is used during
+* the pre rendering at build time for dynamic route.
+* For now we want to prerender all the video session dynamic routes
+* More over netlify lower case all the urls so if we try to access directly to a
+* session video on the address /sessions/rT0FUs7uUks netlify will redirect to /sessions/rt0fus7uuks
+* And then our netlify function will not be able to fetch the right video ID.
+* This js script also generates a mapper `case-sensitive-video-id-mapper.js` located in `src/functions/utils`
+* To map lower case video ids -> case sensitive video Ids
+* https://answers.netlify.com/t/my-url-paths-are-forced-into-lowercase/1659 */
+const fetch = require("node-fetch");
+const {ensureFileSync, writeFileSync} = require('fs-extra');
+const DIST_DYNAMIC_ROUTES_FILENAME = 'routes.txt';
+const DIST_MAPPER_JS_FILE = 'src/functions/utils/case-sensitive-video-id-mapper.js';
+console.log('Generating routes.txt for dynamic routes pre rendering');
+
+const getDistFilename = () => {
+  return DIST_DYNAMIC_ROUTES_FILENAME;
+}
+
+const getMapperFile = () => {
+  return DIST_MAPPER_JS_FILE;
+}
+
+let videoIds = [];
+
+/**
+ * Fetch the playlist by id and map the response to return only the ids
+ * @returns {Promise<void>}
+ */
+const getVideoIds = async () => {
+  // If video ids already computed return them
+  if(videoIds && videoIds.length > 0) {
+    return new Promise(resolve => resolve(videoIds));
+  }
+  // Fetch the playlist of ngx-darija
+  return fetch('http://localhost:8889/.netlify/functions/playlist')
```

Comment:
not against .then but we'd better stick to one syntax, and `await` has already been used in other files.